zerver/lib/bot_config.py

```diff
@@ -0,0 +1,24 @@
+from django.conf import settings
+from zerver.models import BotUserConfigData, UserProfile
+
+from typing import Text, Dict
+
+class ConfigError(Exception):
+    pass
+
+def get_bot_config(bot_profile):
+    # type: (UserProfile) -> Dict[Text, Text]
+    entries = BotUserConfigData.objects.filter(bot_profile=bot_profile)
+    return {entry.key: entry.value for entry in entries}
+
+def set_bot_config(bot_profile, key, value):
+    # type: (UserProfile, Text, Text) -> None
+    entry_size_limit = settings.BOT_CONFIG_ENTRY_SIZE_LIMIT
+    if len(value) > entry_size_limit:
```

Comment:
Do we want this to be a total size, like the storage is?  I think that's a better model...

Reply:
My thinking was that the number of entries cannot be altered by the user, so it's no risk to set an entry-based limit, and it will be easier for the admin to add a bot that requires exceptionally much data (simply make it have more keys). But I guess that is rather unlikely, so will keep it consistent and have a total size.